File: sample/sample-android/src/main/java/com/example/sampleandroid/view/model/ApiSingleScreen.kt
package com.example.sampleandroid.view.model

import androidx.compose.foundation.layout.Column
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import com.example.sampleandroid.view.widget.SampleTextField
import kim.jeonghyeon.androidlibrary.compose.Screen
import kim.jeonghyeon.androidlibrary.compose.unaryPlus
import kim.jeonghyeon.androidlibrary.compose.widget.Button
import kim.jeonghyeon.androidlibrary.extension.resourceToString
import kim.jeonghyeon.sample.compose.R
import kim.jeonghyeon.sample.viewmodel.ApiSingleViewModel

@Composable
fun ApiSingleScreen(model: ApiSingleViewModel) {
    Screen(model) {
        Column {
            Text("current value : ${+model.result}")
            SampleTextField("Input value", model.input)
            Button("update") { model.onClick() }
        }
    }
}

File: sample/sample-native/kotlinIOS/screen/model/ApiSingleScreen.swift
//
// Created by hyun kim on 04/07/20.
// Copyright (c) 2020 hyun kim. All rights reserved.
//

import Foundation
import SwiftUI
import sample_base


func ApiSingleScreen(_ model: ApiSingleViewModel) -> some View {
    Screen(model) {
        Column(alignment: .center) {
            Text("current value : \(model.result.asValue(viewModel: model) ?? "")")
            SampleTextField("Enter value", model.input).frame(width: 100, alignment: .center)
            Button("Update") { model.onClick() }
        }
    }
}
